### PR TITLE
To resolve #9 - TCOP structure

### LINKED ISSUE #9: TCOP: Create TCOP skeleton - Identifying headings/ subheadings

### PR DESCRIPTION
**Pull request by timgent**
Putting in basic structure for TCOP (technical code of practice) based on discussion with Gabby, Abdul, and Tim

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+When contributing to this repository, please first discuss the change you wish to make via issue,
+email, or any other method with the owners of this repository before making a change.
+
+Please note we have a code of conduct, please follow it in all your interactions with the project.
+
+## Pull Request Process
+
+1. We recommend raising an issue first to discuss the changes you are interested in (though please skip this step if the changes are straightforward)
+2. Create your Pull Request
+3. You may merge the Pull Request in once you have the sign-off of two other developers, or if you
```
Comment (chrisns) [inline CONTRIBUTING.md]: do you want to maybe qualify that its not developers that should merge it, perhaps members of the CoE?

File: README.md [hunk 2]
```diff
@@ -1,20 +1,34 @@
+# Home Office Technology Code of Practice
+TODO: Summary of what the TCOP does in brief. Should include it's purpose and how it related to the GDS Service Standards and the GDS Technology Code of Practice.
```
Comment (danielepolencic) [inline README.md]: I think it's `its purpose`.

File: README.md [hunk 3]
```diff
@@ -1,20 +1,34 @@
...
+
+We recommend everyone involved in technology projects understand this guidance.
+
+## How services are assessed
+
+### Periodic assessments
+All services will be [assessed](/assessment.md) against these standards and principles.
+
+### Realtime metrics
+We will be encouraging projects to provide some standard realtime metrics to enable an overall view of our IT estate and how it is performing. These are still to be determined and we welcome input from teams.
```
Comment (chrisns) [inline README.md]: maybe give an email address or a make an issue/slack to receive this input?

File: README.md [hunk 4]
```diff
@@ -1,20 +1,34 @@
...
+
+## Technical best practices and guidance
+Want to meet our standards and pass your assessments first time? Following the [best practices](/best-practices/README.md) here will go a long way to helping you do this.
 
-## GDS service standard and code of practice
```
Comment (chrisns) [inline README.md]: services don't need to meet the GDS standard now?

File: README.md (hunk 3 above)
Comment (danielepolencic) [inline README.md]: @shabe mentioned that there's a gov wide effort to publish metrics to https://data.gov.uk/ ?

File: tech-stack.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
+# Tech Stack
+
+## Selecting languages, frameworks and technology
+The main languages we recommend for application development are:
+
+- Javascript
+- Java
+- Scala
+
+For more detail on recommended technologies please see the [complete tech stack](https://trello.com/b/IGRXc88M/technology-stack-beta). For access please [contact the Centre of Excellence](mailto: coe-eng-arch@digital.homeoffice.gov.uk).
```
Comment (chrisns) [inline tech-stack.md]: do we want to promote the trello given that `Microsoft Windows Server 2016` is `Investing (Support Growing)`?

File: README.md (hunk 3 above)
Comment (chrisns) [inline README.md]: we'd need to figure out how we want to do that relationship, e.g. do we capture it, host it and publish on data.gov.uk or do we make every team do that work and then consume it like any other external body/person could

File: CONTRIBUTING.md (hunk 1 above)
Comment (timgent) [inline CONTRIBUTING.md]: Agree, will update

File: README.md (hunk 3 above)
Comment (timgent) [inline README.md]: Yep agree, will make an issue and link to it here

File: tech-stack.md (hunk 5 above)
Comment (timgent) [inline tech-stack.md]: I think we want to promote the Trello regardless, but we need to revisit what's on there and how we maintain the board. In my view we'd be better off transferring it to Github to make it easier to manage and debate things like this

File: README.md (hunk 2 above)
Comment (timgent) [inline README.md]: Yep, will fix, thanks. I'll get the hang of those pesky apostrophes one day :)

File: README.md (hunk 3 above)
Comment (timgent) [inline README.md]: I've created an issue here where we can discuss this stuff more:
https://github.com/UKHomeOffice/engineering-and-architecture/issues/11

**Comment by danielepolencic** [comment]
LGTM

**Comment by timgent** [comment]
Discussed with @gabbym54 we think this is ready to merge... once @chrisns approves :)

**Comment by gabbym54** [comment]
@timgent - I am going to create placeholder files for the 'missing' links within the TCoP, e.g tcop-background.md.  Do I create them on the master or on a new branch with a new pull request? :)

**Comment by chrisns** [comment]
@gabbym54 it'll need to go via a new branch and PR, master is protected

**Pull request merged**